### PR TITLE
Ideomatic Go interface for log exporter

### PR DESCRIPTION
**Pull request by furkhat**

### CODE REVIEW

File: castai/client.go [hunk 1]
```diff
@@ -23,11 +24,19 @@ const (
...
+type client struct {
+	log       *logrus.Logger
+	rest      *resty.Client
+	clusterID string
+}
+
+// client must satisfy logexporter.Sender.
+var _ logexporter.Sender = new(client)
+
+func NewClient(log *logrus.Logger, rest *resty.Client, clusterID string) *client {
```
Comment (ValyaB) [inline castai/client.go:39]: Public func returns a public struct
make client public

Comment (stgleb) [reply castai/client.go:39]: type itself yes, but fields are still private, aren't they?

File: logexporter/exporter.go [hunk 2]
```diff
@@ -20,20 +18,34 @@ type Exporter interface {
 	Wait()
 }
 
-func NewExporter(logger *logrus.Logger, client castai.Client) Exporter {
-	return &exporter{
-		logger: logger,
-		client: client,
-		wg:     sync.WaitGroup{},
-	}
```
Comment (stgleb) [inline logexporter/exporter.go:28]: Maybe LogSender than?

**Review by stgleb — APPROVED** [review]
LGTM

File: castai/client.go (hunk 1 above)
Comment (furkhat) [reply castai/client.go:39]: Yep I will make the type public and fields will stay private.

File: logexporter/exporter.go (hunk 2 above)
Comment (furkhat) [reply logexporter/exporter.go:28]: I think linter was complaining that there is a tautology. It was saying that `logexporter.Sender` is better than `logexporter.LogSender`.
I will double check and change if it doesn't complain.

**Pull request merged**